### PR TITLE
Add config url trailing slash

### PR DESCRIPTION
**Pull request by tdugue**
### Related Issues

following the opening of this issue (https://github.com/DivanteLtd/vue-storefront/issues/4684) I created a configuration ( urlTrailingSlash ) to add or not the slash at the end of a url.

I changed the normalizeUrlPath function so that depending on the configuration the slash is added or not.

I also changed the prepareDynamicRoute function and added the pathToRegexpOptions attribute with the strict option following the slash config. pathToRegexpOptions tells the vuejs router not to remove the slash.

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by tdugue** [comment]
Hi,

After some test there are still problems the code does not work for all cases.

I try to update the PR quickly

Thanjs

**Comment by gibkigonzo** [comment]
@tdugue can we do that without config? :thinking: I think it should work by default, wdyt?

File: core/modules/url/helpers/index.ts [hunk 1]
```diff
@@ -57,7 +62,8 @@ export function findRouteByPath (path: string): RouteConfig {
 export function normalizeUrlPath (url: string): string {
   if (url && url.length > 0) {
     if (url.length > 0 && !url.startsWith('/')) url = `/${url}`
-    if (url.endsWith('/')) url = url.slice(0, -1)
+    if (url.endsWith('/') && !config.seo.urlTrailingSlash) url = url.slice(0, -1)
+    if (!url.endsWith('/') && config.seo.urlTrailingSlash) url += '/'
     const queryPos = url.indexOf('?')
     if (queryPos > 0) url = url.slice(0, queryPos)
   }
```
Comment (gibkigonzo) [inline core/modules/url/helpers/index.ts:69]: Last changes in different file:
1. https://github.com/DivanteLtd/vue-storefront/blob/master/core/modules/url/router/beforeEach.ts#L26
change `const path = normalizeUrlPath(to.path)` to `const path = normalizeUrlPath(to.path, false)` => in this place we removed trailing slash, so we need to use `normalizeUrlPath` with `false` which will result in keeping trailing slash
2. https://github.com/DivanteLtd/vue-storefront/blob/master/core/modules/url/router/beforeEach.ts#L33
change `if (routeData) {` to `if (routeData && !routeData.name.endsWith('page-not-found')) {`

File: config/default.json [hunk 2]
```diff
@@ -92,6 +92,7 @@
   "seo": {
     "useUrlDispatcher": true,
     "disableUrlRoutesPersistentCache": true,
+    "urlTrailingSlash": false,
```
Comment (gibkigonzo) [inline config/default.json:95]: :gun: remove this :pray:

**Comment by gibkigonzo** [comment]
@tdugue please rebase this branch to `hotfix/v1.12.3` and update changelog in v1.12.3 section. Also I've added comments what should be changed to make it work. :pray: (you may need to update test cases, but I'm not sure)

File: core/modules/url/helpers/index.ts [hunk 3]
```diff
@@ -57,7 +62,8 @@ export function findRouteByPath (path: string): RouteConfig {
 export function normalizeUrlPath (url: string): string {
   if (url && url.length > 0) {
     if (url.length > 0 && !url.startsWith('/')) url = `/${url}`
-    if (url.endsWith('/')) url = url.slice(0, -1)
+    if (url.endsWith('/') && !config.seo.urlTrailingSlash) url = url.slice(0, -1)
+    if (!url.endsWith('/') && config.seo.urlTrailingSlash) url += '/'
```
[the remark this answers is not in the document]
Comment (gibkigonzo) [reply core/modules/url/helpers/index.ts:66]: this change probably require update in tests :point_up:

**Comment by Fifciu** [comment]
Due to the author's inactivity, I decided to finish this PR on my own: https://github.com/vuestorefront/vue-storefront/pull/5372

**Pull request closed without merging**